daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs

```diff
@@ -388,10 +389,42 @@ runLedgerGetDalfs lflags pkgIds exclPkgIds
 runLedgerListPackages :: LedgerFlags -> IO [LF.PackageId]
 runLedgerListPackages lflags = do
     args <- getDefaultArgs lflags
-    runWithLedgerArgs args $ do
-        lid <- L.getLedgerIdentity
-        pkgIds <- L.listPackages lid
-        pure [LF.PackageId $ TL.toStrict $ L.unPackageId pid | pid <- pkgIds]
+    case api args of
+      Grpc ->
+        runWithLedgerArgs args $ do
+            lid <- L.getLedgerIdentity
+            pkgIds <- L.listPackages lid
+            pure [LF.PackageId $ TL.toStrict $ L.unPackageId pid | pid <- pkgIds]
+      HttpJson -> httpJsonRequest args "GET" "/v1/packages" id
+
+-- | Same as runLedgerListPackages, but print output.
+runLedgerListPackages0 :: LedgerFlags -> IO ()
+runLedgerListPackages0 flags = do
+    pkgs <- runLedgerListPackages flags
+    rdalfs <- runLedgerGetDalfs flags pkgs []
+    putStrLn "Available packages: "
+    putStrLn $
+        unlines $
+        map T.unpack $
+        [ LF.unPackageName name <> "-" <> LF.unPackageVersion version <> " : " <>
+        LF.unPackageId remoteDalfPkgId
+        | RemoteDalf {..} <- rdalfs
+        , Just name <- [remoteDalfName]
+        , Just version <- [remoteDalfVersion]
+        ] ++
+        [ LF.unPackageName name <> "-" <> "<unknown version>" <> " : " <>
+        LF.unPackageId remoteDalfPkgId
+        | RemoteDalf {..} <- rdalfs
+        , Just name <- [remoteDalfName]
+        , Nothing <- [remoteDalfVersion]
+        ]
+    putStrLn "Available packages without package name/version info:"
```

Comment:
Instead of having two separate lists maybe just use something like the following?
```
pkgid1
pkgid2 (name-version)
pkgid3
…
```

Reply:
changed it to this format.